packages/gatsby/cache-dir/loader.js

```diff
@@ -359,6 +363,20 @@ const queue = {
       return null
     }
   },
+
+  fetchPageHtml: rawPath => {
```

Comment:
`doesPageExist` is much better.

Agree on `HEAD` too. I was hoping that by doing a full GET request, the browser would be able to use that cached version when it reloaded. But html isn't cached at all so it will have to do two requests anyway.

Reply:
Why wouldn't it be cached? It all depends on the settings of the server, right? If the server has proper cache-control headers for HTML it should cache it? (or is it browser behaviour?

Reply:
My understanding is that html files are normally uncached whereas other assets are. Since html files aren't content hashed. Either way, I think `HEAD` makes more sense since the main purpose is to find out whether the html file exists or not, so we know whether to show a 404.